bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,616 @@
+                  `in this breaking changes file:\n\n` +
+                  `'${breakingChangesFile}'\n\n` +
+                  'Parsed YAML:\n\n' +
+                  `${JSON.stringify(safeHistory, null, 4)}\n\n` +
+                  'API history block:\n\n' +
+                  `${possibleHistoryBlock.value}\n\n`,
+              );
+              errorCounter++;
+            }
+          }
+        }
+
+        if (checkPullRequestLinks) {
+          const allPrReleaseVersions = await getAllPrReleaseVersions();
+
+          const safeHistory = unsafeHistory as ApiHistory;
+          const prsInHistory: Array<string> = [];
+
+          // Copied from <https://github.com/electron/website/blob/a5d30f1ede6b20ea00d487c198c71560745063ab/src/transformers/api-history.ts#L154-L174>
+          safeHistory.added?.forEach((added) => {
+            prsInHistory.push(added['pr-url'].split('/').at(-1)!);
+          });
+
+          safeHistory.changes?.forEach((change) => {
+            prsInHistory.push(change['pr-url'].split('/').at(-1)!);
+          });
+
+          safeHistory.deprecated?.forEach((deprecated) => {
+            prsInHistory.push(deprecated['pr-url'].split('/').at(-1)!);
+          });
+
+          for (const prNumber of prsInHistory) {
+            if (!allPrReleaseVersions.hasOwnProperty(Number(prNumber))) {
+              // ? Should this be an error or warning?
+              console.warn(
+                'Warning occurred while parsing Markdown document:\n\n' +
+                  `'${filepath}'\n\n` +
+                  `Couldn't find PR number:\n\n` +
+                  `'${prNumber}'\n\n` +
+                  'in list of all PRs included in releases.\n\n' +
+                  `Maybe the list is stale? Are you documenting a new change?\n\n` +
+                  'Parsed YAML:\n\n' +
+                  `${JSON.stringify(safeHistory, null, 4)}\n\n` +
+                  'API history block:\n\n' +
+                  `${possibleHistoryBlock.value}\n\n`,
+              );
+              warningCounter++;
+            }
+          }
+        }
+
+        // ? Maybe collect all api history and export it to a single file for future use.
+      }
+
+      // ? Maybe replace user YAML with result of <https://eemeli.org/yaml/#tostring-options> for consistent style (but not in CI)
+    }
+  } catch (error) {
+    errorCounter++;
+    console.error('Error occurred while linting:\n', error);
+  } finally {
+    return { historyBlockCounter, documentCounter, errorCounter, warningCounter };
+  }
+}
+
+function parseCommandLine() {
+  const showUsage = (arg?: string): boolean => {
+    if (!arg || arg.startsWith('-')) {
+      console.log(
+        'Usage: lint-roller-markdown-api-history [--root <dir>] <globs>' +
+          ' [-h|--help]' +
+          ' [--check-placement] [--check-pull-request-links] [--breaking-changes-file <path>] [--check-strings]' +
+          ' [--schema <path>]' +
+          ' [--ignore <globs>] [--ignore-path <path>]',
```

Comment:
comment: this is out of scope of the PR, but I feel like lint-roller's various usage prompts would benefit from having one-liner descriptions since something like `--check-placement` is not super clear until you read the source code.

cc @dsanders11

Reply:
It's hard to come up with flag names that are descriptive enough but also not 40 characters long 😭 . <sup>[[xkcd#1168](https://xkcd.com/1168/)] [[xkcd#1597](https://xkcd.com/1597/)]</sup>

Time for a man page?

<img width="200px" src="https://github.com/electron/lint-roller/assets/99439005/1571d504-b971-4e2e-974f-ecc6c947f553"></img>